go/cmd/dolt/commands/show.go

```diff
@@ -186,181 +321,255 @@ func parseHashString(hashStr string) (hash.Hash, error) {
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("error getting branches for hash '%s': %v", commitHash, err)
+	}
+	remoteBranchesForHash, err := getBranchesForHash(queryist, sqlCtx, commitHash, false)
+	if err != nil {
+		return nil, fmt.Errorf("error getting remote branches for hash '%s': %v", commitHash, err)
+	}
+	tagsForHash, err := getTagsForHash(queryist, sqlCtx, commitHash)
+	if err != nil {
+		return nil, fmt.Errorf("error getting tags for hash '%s': %v", commitHash, err)
 	}
 
-	datasets := &diffDatasets{
-		fromRoot: parentRoot,
-		toRoot:   commitRoot,
-		fromRef:  parentHash.String(),
-		toRef:    cmHash.String(),
+	ci := &commitInfo{
+		commitMeta: &datas.CommitMeta{
+			Name:          name,
+			Email:         email,
+			Timestamp:     timestamp,
+			Description:   message,
+			UserTimestamp: int64(timestamp),
+		},
+		commitHash:        commitHash,
+		height:            height,
+		isHead:            isHead,
+		localBranchNames:  localBranchesForHash,
+		remoteBranchNames: remoteBranchesForHash,
+		tagNames:          tagsForHash,
 	}
 
-	// An empty string will cause all tables to be printed.
-	var tableNames []string
+	if parent != "" {
+		ci.parentHashes = []string{parent}
+	}
 
-	tableSet, err := parseDiffTableSet(ctx, dEnv, datasets, tableNames)
+	return ci, nil
+}
+
+func getTagsForHash(queryist cli.Queryist, sqlCtx *sql.Context, targetHash string) ([]string, error) {
+	q := fmt.Sprintf("select tag_name from dolt_tags where tag_hash = '%s'", targetHash)
+	rows, err := getRowsForSql(queryist, sqlCtx, q)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	dArgs := &diffArgs{
-		diffDisplaySettings: opts.diffDisplaySettings,
-		diffDatasets:        datasets,
-		tableSet:            tableSet,
+	tags := []string{}
+	for _, row := range rows {
+		name := row[0].(string)
+		tags = append(tags, name)
+	}
+	return tags, nil
+}
+
+func getBranchesForHash(queryist cli.Queryist, sqlCtx *sql.Context, targetHash string, getLocalBranches bool) ([]string, error) {
+	var q string
+	if getLocalBranches {
+		q = fmt.Sprintf("select name, hash from dolt_branches where hash = '%s'", targetHash)
+	} else {
+		q = fmt.Sprintf("select name, hash from dolt_remote_branches where hash = '%s'", targetHash)
+	}
+	rows, err := getRowsForSql(queryist, sqlCtx, q)
+	if err != nil {
+		return nil, err
 	}
 
-	return diffUserTables(ctx, dEnv, dArgs)
+	branches := []string{}
+	for _, row := range rows {
+		name := row[0].(string)
+		branches = append(branches, name)
+	}
+	return branches, nil
+}
+
+func getHashOf(queryist cli.Queryist, sqlCtx *sql.Context, ref string) (string, error) {
+	q := fmt.Sprintf("select hashof('%s')", ref)
+	rows, err := getRowsForSql(queryist, sqlCtx, q)
+	if err != nil {
+		return "", err
+	}
+	if len(rows) == 0 {
+		return "", fmt.Errorf("no commits found for ref %s", ref)
+	}
+	return rows[0][0].(string), nil
+}
+
+func getTimestampColAsUint64(col interface{}) (uint64, error) {
```

Comment:
Is this going to be useful to other cli commands? Perhaps move to utils.go? Up to you, but I'd like to make sure we don't forget this is here.

Reply:
Moving to utils.go